src/app/services/covid-data-api.service.ts

```diff
@@ -31,37 +29,20 @@ export class CovidDataApiService implements CovidDataApiServiceInterface {
    * Calls all api services and returns the first non-error response
    * If all lead to errors, returns an empty observable
    *
-   * e.g. to call the 'foo' method of each subservice with args 1, 2, 3 call as
-   * `this.getResultFromOneApiFor(this.foo, 1, 2, 3)`
-   *
-   * Only works if this method and each subservice implement the same interface/functions
-   *
-   * @param {Observable<T>} method The method to be called
+   * @param {Observable<CovidDataPoint>} method The method to be called
    */
-  private getResultFromOneApiFor<T>(
-    method: (...args: any[]) => Observable<T>, // reference to a method with the same name being called
-    ...args: any[]
-  ): Observable<T> {
-    // if the method is not a class method of this class throw an error
-    if (
-      !Object.getOwnPropertyNames(CovidDataApiService.prototype).includes(
-        method.name
-      )
-    ) {
-      throw new Error(
-        `Method: ${method.name}() not found in Object: ${this.constructor.name}`
-      );
-    }
-    let errorCount = 0; // keep track of number or errored out requests for later
-    return race<T>( // only need result from one of the apis
-      this.apiServices.map<Observable<T>>((s, i, arr) => 
+  private getResultFromOneApiFor(country?: Country): Observable<CovidDataPoint> {
+    // TODO fix race condition, possibly using reduce?
+      let errorCount = 0; // keep track of number or errored out requests for later
+    return race( // only need result from one of the apis
+      this.apiServices.map<Observable<CovidDataPoint>>((api, i, arr) => 
         // call the wanted method from each api
-        s[method.name](...args).pipe( 
+        new GetLatestDataCommand(api, country).call().pipe( 
```

Comment:
One more nitpick, if you ever want to use a different command, you'll have to rewrite the entire function.
This seems to defeat the purpose of the command pattern, because you might as well just put an if statement here.
You probably want to have something like injecting a list of commands instead of a list of apis.

Reply:
I think I've got an idea (not sure if its a good idea yet) for this and will add another small PR with it